app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Katello
+  class Api::Rhsm::CandlepinDynflowProxyController < Api::V2::ApiController
+    include Katello::Authentication::ClientAuthentication
+    include ForemanTasks::Triggers
+    include AbstractController::Callbacks
+
+    before_action :find_host, :only => [:upload_package_profile]
+    before_action :authorize_client_or_user, :only => [:upload_package_profile]
+
+    #api :PUT, "/consumers/:id/packages", N_("Update installed packages")
+    #api :PUT, "/consumers/:id/profile", N_("Update installed packages")
+    #param :id, String, :desc => N_("UUID of the consumer"), :required => true
+    def upload_package_profile
+      User.as_anonymous_admin do
+        async_task(::Actions::Katello::Host::UploadPackageProfileWrapper, @host, request.body.string)
+      end
+      [
+        200,
+        { "Content-Type" => "application/json" },
+        Resources::Candlepin::Consumer.get(@host.subscription_facet.uuid)
+      ]
+    end
```

Comment:
It depends if the actions are executed during the test. If they are not executed, then we could count total number of allocations delta (which should go down significally). If the actions are performed, then we have a problem - I'm just moving the allocations into the action, so they will still be counted.